- Fix building my_gethwaddr() on OpenBSD
  899cedb33c8e92a0067f2d74015300454c6235ae
- Fix building my_gethwaddr() on OpenBSD - part for 10.5 and newer
  ab81aefef6e54ec872622f996ed66626700e4d8e

Index: mysys/my_gethwaddr.c
--- mysys/my_gethwaddr.c.orig
+++ mysys/my_gethwaddr.c
@@ -23,7 +23,7 @@
 
 #ifndef MAIN
 
-#if defined(_AIX) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__linux__) || defined(__sun) || defined(_WIN32)
+#if defined(_AIX) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__linux__) || defined(__sun) || defined(_WIN32)
 static my_bool memcpy_and_test(uchar *to, uchar *from, uint len)
 {
   uint i, res= 1;
@@ -35,8 +35,14 @@ static my_bool memcpy_and_test(uchar *to, uchar *from,
 }
 #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#ifdef __OpenBSD__
+#include <netinet/in.h>
+#include <net/if_arp.h>
+#include <netinet/if_ether.h>
+#else
 #include <net/ethernet.h>
+#endif
 #include <sys/sysctl.h>
 #include <net/route.h>
 #include <net/if.h>
